src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+* @file SignalGenerator.cpp
+* @brief This header file contains the implementation functions of the SignalGenerators class.
+*
+* @author Murilo Mendonça Venâncio
+* 
+* @date 24/01/2019
+*/
+
+#include "SignalGenerator.hpp"
+
+
+SignalGenerator::~SignalGenerator()
+{
+
+}
+
+bool SignalGenerator::setAmplitude(const double amp)
+{
+  this->m_amp    = amp;
```

Comment:
@murilomend, please avoid to this. I usually just include tabs like here, when I want to align the operator **"="**.

I also avoid to use `this` inside the class. Only when it is required use `this`.

Please, replace this for:

```
m_amp = amp;
```
Please, also apply this modifications in the other parts of the code.